x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/OperatorUserDescriptor.java

```diff
@@ -0,0 +1,274 @@
+            }
+            if (authenticationType != null) {
+                sb.append(", auth_type=").append(authenticationType.name().toLowerCase(Locale.ROOT));
+            }
+            sb.append("]");
+            return sb.toString();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o)
+                return true;
+            if (o == null || getClass() != o.getClass())
+                return false;
+            Group group = (Group) o;
+            return usernames.equals(group.usernames) && Objects.equals(realmName,
+                group.realmName) && realmType.equals(group.realmType) && authenticationType == group.authenticationType;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(usernames, realmName, realmType, authenticationType);
+        }
+    }
+
+    public static List<Group> parseFileLenient(Path path, Logger logger) {
+        if (false == Files.exists(path)) {
+            logger.debug("Skip reading operator user file since it does not exist");
+            return List.of();
+        }
+        logger.debug("Reading operator users file [{}]", path.toAbsolutePath());
+        try {
+            return parseFile(path);
+        } catch (IOException | RuntimeException e) {
+            logger.error("Failed to parse operator_users file [" + path + "].", e);
+            return List.of();
+        }
+    }
+
+    public static List<Group> parseFile(Path path) throws IOException {
+        try (InputStream in = Files.newInputStream(path, StandardOpenOption.READ)) {
+            return parseConfig(in);
+        }
+    }
+
+    public static List<Group> parseConfig(InputStream in) throws IOException {
+        final List<Group> groups = new ArrayList<>();
+        try (XContentParser parser = yamlParser(in)) {
+            XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+            while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+                XContentParserUtils.ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser);
+                final String categoryName = parser.currentName();
+                if (false == AuthenticationField.PRIVILEGE_CATEGORY_VALUE_OPERATOR.equals(categoryName)) {
+                    throw new IllegalArgumentException("Operator user config file must begin with operator, got [" + categoryName + "]");
+                }
+                XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.nextToken(), parser);
+                while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                    groups.add(parseOneGroup(parser));
+                }
+            }
+        }
+        return List.copyOf(groups);
+    }
+
+    private static Group parseOneGroup(XContentParser parser) throws IOException {
```

Comment:
Is it not possible to do this with `ObjectParser`?

Reply:
It is possible. I was just more comfortable with manual parsing. Anyway, I changed it to be `ObjectParser`.